beacon-chain/rpc/prysm/v1alpha1/validator/proposer.go

```diff
@@ -404,3 +425,58 @@ func (vs *Server) SubmitValidatorRegistrations(ctx context.Context, reg *ethpb.S
 
 	return &emptypb.Empty{}, nil
 }
+
+func (vs *Server) validateConsensus(ctx context.Context, blk interfaces.ReadOnlySignedBeaconBlock) error {
+	parentState, err := vs.StateGen.StateByRoot(ctx, blk.Block().ParentRoot())
+	if err != nil {
+		return errors.Wrap(err, "could not get parent state")
+	}
+	_, err = transition.ExecuteStateTransition(ctx, parentState, blk)
+	if err != nil {
+		return errors.Wrap(err, "could not execute state transition")
+	}
+	return nil
+}
+
+func (vs *Server) validateEquivocation(blk interfaces.ReadOnlyBeaconBlock) error {
+	if vs.seenProposerIndex(blk.Slot(), blk.ProposerIndex()) {
```

Comment:
no longer needs to use the cache here, just have to check if there exists a block of same slot in forkchoice store